### DISCUSSION TITLE
How to use custom breakpoints

### DISCUSSION THREAD

**Discussion by mvrcwest** [msberends/AMR#37] [Q&A] [answered]
I did not think this belonged under issues, so I am writing here instead.
The eucast_custom_rules function does not support "reversing" rules, which have already been applied, it is more like additional rules to add. And since there were some minor mistakes, and because we deviate at our lab from some specific rules due to say using po breakpoints for some antibiotics, I got my hands dirty and tried modifying the eucast_rules.tsv - The changes I made in "breakpoints" seem to have worked, but what bothers me is that some expert rules (the ones on seperate eucast tabs/species) still gets applied even though I removed them. So the big question now is if there are any other files storing these rules, which I need to modify in order for my changes to work?
When I have the time I will make a pull request for the minor mistakes I've encountered, or at least open an issue on them.
Thank you for your time
Marc Westerholt

**Comment by msberends** [comment, marked as answer]
Oof, that's not the most convenient way 🙂 

But it is already possible. Although I'm not entirely sure where you're aiming at. There are two forms of EUCAST rules, their expert rules and their interpretation of MIC/disk values.

## For EUCAST expert rules

The problem for you is that we save the TSV file to an R format with every update we make. That process does not happen for you locally. Fortunately, you have two options:

1. Use `eucast_rules(rules = "custom")`. This is the recommended way and will omit all official rules and only apply your rules. Downside is that you would have to write all EUCAST rules you want to keep yourself manually into the `custom_eucast_rules()` function.
2. Use a hack. The `eucast_rules()` function accepts the undocumented argument `eucast_rules_df` as you can see here: 
   https://github.com/msberends/AMR/blob/fa2f5214b9703c8410792fbdf2e2f57badaf90d0/R/eucast_rules.R#L561-L569
  
    So if you transform the original EUCAST file and save it as a `data.frame`, you can use it for input in the `eucast_rules()` function. The implemented EUCAST file is internally available as `AMR::: eucast_rules_file`:

   ```r
   library(dplyr)
   library(AMR)

   all_custom <- AMR::: eucast_rules_file %>%
    # no Staphylococcus rules
    filter(reference.version == 11 & !this_value %like% "Staphylococcus")
 
    eucast_rules(eucast_rules_df = all_custom)
   ```

   Didn't test it though, but it should work.

## For EUCAST interpretation of MIC/disk diffusion values

For this, the `as.rsi()` function must be used. It has an argument `reference_data` which defaults to `AMR::rsi_translation`. [The documentation says about this](https://msberends.github.io/AMR/reference/as.rsi.html#arguments):

> a data.frame to be used for interpretation, which defaults to the [rsi_translation](https://msberends.github.io/AMR/reference/rsi_translation.html) data set. Changing this argument allows for using own interpretation guidelines. This argument must contain a data set that is equal in structure to the [rsi_translation](https://msberends.github.io/AMR/reference/rsi_translation.html) data set (same column names and column types). Please note that the `guideline` argument will be ignored when `reference_data` is manually set.

So if you want to edit the breakpoint for oral use, you should make changes to the rsi_translation data set and use something like:

```r
as.rsi(reference_data = mycustombreakpoints)

my_data %>%
  mutate_if(is.mic, as.rsi, reference_data = mycustombreakpoints)
```

Does this answer your question?

**Reply by mvrcwest** [reply]
Hi Matthijs
Thank you for a very elaborate and educating reply.
I went with option 1, I am not as tech savvy as you are, and it seemed as the most sustainable solution.
I reckon many will use the custom_eucast_rules function, and therefore suggest enabling multiple antibiotics after the tilde. 
Again thank you for your help, it was much appreciated.
Marc Westerholt
